File: algorithms/easy/sum_multiples.ex
defmodule Solution do
  @spec sum_of_multiples(n :: integer) :: integer
  def sum_of_multiples(n) do
    1..n
    |> Enum.filter(fn n ->
      [3, 5, 7]
      |> Enum.any?(fn m -> rem(n, m) == 0 end)
    end)
    |> Enum.sum()
  end
end

File: algorithms/easy/sum_multiples.rs
impl Solution {
    pub fn sum_of_multiples(n: i32) -> i32 {
        let mut sum = 0;
        let arr = [3, 5, 7];
        for i in (1..=n) {
            if arr.iter().any(|&j| i % j == 0) {
                sum += i;
            }
        }
        sum
    }
}
